improver_tests/regrid/test_RegridWithLandSeaMask.py

```diff
@@ -334,3 +336,63 @@ def test_regrid_bilinear_with_mask_2():
     np.testing.assert_allclose(
         regrid_bilinear_with_mask.data, expected_results, atol=1e-3
     )
+
+
+@pytest.mark.parametrize("regridder", ("nearest", "bilinear"))
+@pytest.mark.parametrize("landmask", (True, False))
+@pytest.mark.parametrize("maskedinput", (True, False))
+def test_target_domain_bigger_than_source_domain(regridder, landmask, maskedinput):
+    """Test regridding when target domain is bigger than source domain"""
+
+    # set up source cube, target cube and land-sea mask cube
+    cube_in, cube_out_mask, cube_in_mask = define_source_target_grid_data_same_domain()
+
+    # add a circle of grid points so that output domain is much bigger than input domain
+    width_x, width_y = 2, 4  # lon,lat
+    cube_out_mask_pad = pad_cube_with_halo(cube_out_mask, width_x, width_y)
+
+    if landmask:
+        with_mask = "-with-mask"
+    else:
+        with_mask = ""
+        cube_in_mask = None
+    regrid_mode = f"{regridder}{with_mask}-2"
+
+    if maskedinput:
+        # convert the input data to a masked array with no values covered by the mask
+        cube_in_masked_data = np.ma.masked_array(cube_in.data, mask=False)
+        cube_in.data = cube_in_masked_data
+
+    # run the regridding
+    regrid_out = RegridLandSea(
+        regrid_mode=regrid_mode, landmask=cube_in_mask, landmask_vicinity=250000000,
+    )(cube_in, cube_out_mask)
+
+    regrid_out_pad = RegridLandSea(
+        regrid_mode=regrid_mode, landmask=cube_in_mask, landmask_vicinity=250000000,
+    )(cube_in, cube_out_mask_pad)
+
+    # check that results inside the padding matches the same regridding without padding
+    np.testing.assert_allclose(
+        regrid_out.data, regrid_out_pad.data[width_y:-width_y, width_x:-width_x],
+    )
+
+    # check results in the padded area
+    # NOTE: there is an extra column with values in the X dimension compared to
+    # the assert_allclose checking the data above
+    if maskedinput:
+        # masked array input should result in masked array output
+        assert hasattr(regrid_out_pad.data, "mask")
+        regrid_out_pad.data.mask[width_y:-width_y, width_x:-width_x] = True
+        np.testing.assert_array_equal(
+            regrid_out_pad.data.mask,
+            np.full_like(regrid_out_pad.data, True, dtype=np.bool),
+        )
+    else:
```

Comment:
I expected the plugin to return a masked array here too, to differentiate between points that had data and points that did not. Using `np.nan` is likely to cause issues in data smoothing modules as the nans tend to proliferate.

Reply:
just keep output/input consistent here. I reckon that the normal input should be masked.